### PR TITLE
Change vote second time #5780 (Time Between Consecutive Votes)

### PR DESCRIPTION
**Pull request by xillibit**
Pull Request for Issue #5780

### CODE REVIEW

**Comment by rich20** [comment]
These both errors are on all forum pages
Notice: Undefined property: stdClass::$user_id in libraries/kunena/access.php on line 203
Notice: Undefined property: stdClass::$category_id in libraries/kunena/access.php on line 204

This error you get if you click on vote (or change vote)
Notice: A non well formed numeric value encountered in libraries/kunena/forum/topic/topic.php on line 2114

**Comment by xillibit** [comment]
I don't have these errors

**Comment by 810** [comment]
I get this one: 
Notice: A non well formed numeric value encountered in C:/Users/Jelle/Documents/gitkraken/Kunena-Forum/src/libraries/kunena/forum/topic/topic.php on line 2114

File: src/libraries/kunena/forum/topic/poll/poll.php [hunk 1]
```diff
@@ -314,7 +314,7 @@ public function getMyTime($user = null)
 			}
 		}
 
-		return (int) $this->mytime[$user->userid];
+		return $this->mytime[$user->userid];
```
Comment (810) [inline src/libraries/kunena/forum/topic/poll/poll.php]: if you revert this change, the error is gone

**Comment by 810** [comment]
i still can switch polls, when set the option to 00:10:10

**Comment by 810** [comment]
for some reason i get:
$date_a = new DateTime($poll->getMyTime());  =  public 'date' => string '2018-04-30 20:18:00.000000'
but on my database = 2018-04-30 21:58:48

File: src/libraries/kunena/forum/topic/poll/poll.php (hunk 1 above)
Comment (810) [inline src/libraries/kunena/forum/topic/poll/poll.php:317]: add the (int) to line 309, then its ok

**Comment by 810** [comment]
@rich20 i have fixed it, could you try the last commit

**Comment by rich20** [comment]
Change vote seems fixed now. You see the change button only after the choosen time period. Previously you have no change button. But we have the next bug now.
